src/SUnit-Core/DefaultExecutionEnvironment.extension.st

```diff
@@ -4,7 +4,7 @@ Extension { #name : #DefaultExecutionEnvironment }
 DefaultExecutionEnvironment >> runTestCase: aTestCase [
 
 	| testEnv |
-	testEnv := TestExecutionEnvironment new.
+	testEnv := aTestCase executionEnvironmentClass new.
```

Comment:
I like the idea to have specific behavior for execution of given test class. 
But this redirection to test is not enough. It works only for single test but full test suite will ignore it:
```Smalltalk
TestSuite>>run: aResult
   TestExecutionEnvironment new beActiveDuring: [...]
```
Tests here will not be running under defined execution environment. All of them will be managed by same environment instance. And it will be inconsistent to single test run.

I would better redirect to test case the configuration of execution environment. So it will be always performed in same way for given test.
But it would require refactoring of TestExecutionEnvironment to reify such configuration which will allow to disable the process management or inject the extra logic (like UI loop).
I imagine this as a special kind of test resources which will live for single test by registering in execution environment. 
But maybe different concept will be better for this purpose.

Reply:
As continue to my idea I would like to be able to avoid code in tests like:
```Smalltalk
file := 'myTestFile' asFileReference.
[file writeStreamDo: [:s | ...].
...
self assert: ...
] ensure: [file ensureDelete].
```
Or
```Smalltalk
server := MyWebserver new.
[server start.
...
self assert: ...
] ensure: [server stop]
```
Currently we are managing such cases by extracting "instantiation" into setUp, storing "resource" in instance variable, and extracting "cleanup" into tearDown. But it is always quite boring work especially when it is just a single test. Sometimes we are even introducing separate test class to correctly manage such resources.
It can be simplified like:
```Smalltalk
self addResource: [file := 'myTestFile' asFileReference] releaseBy: [file ensureDelete]
file writeStreamDo: [:s | ...].
...
self assert:...
```
and 
```Smalltalk
self addResource: [server := MyWebserver new.] releaseBy: [server stop].
server start.
...
self assert:...
```
From this perspective the current process management is automatic registration of such per-test resources where every forked process is a resource.

I do not ask you to implement it. I just want to inspire your thoughts related to this PR work.